app/src/components/features/rules/TestThisRule/components/TestReportsTable/index.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { useCallback, useMemo, useState } from "react";
+import { getAppMode, getCurrentlySelectedRuleData } from "store/selectors";
+import { useSelector } from "react-redux";
+import { Col, Popconfirm, Row, Spin, Table } from "antd";
+import { TestReport } from "../../types";
+import { getFormattedTimestamp } from "utils/DateTimeUtils";
+import { MdOutlineCheckCircle } from "@react-icons/all-files/md/MdOutlineCheckCircle";
+import { IoMdCloseCircleOutline } from "@react-icons/all-files/io/IoMdCloseCircleOutline";
+import { RiDeleteBin6Line } from "@react-icons/all-files/ri/RiDeleteBin6Line";
+import { MdOndemandVideo } from "@react-icons/all-files/md/MdOndemandVideo";
+import { RQButton } from "lib/design-system/components";
+import { redirectToUrl } from "utils/RedirectionUtils";
+import { deleteTestReport } from "../../helpers";
+import { useBottomSheetContext } from "componentsV2/BottomSheet";
+import { LoadingOutlined } from "@ant-design/icons";
+import Logger from "lib/logger";
+import { toast } from "utils/Toast";
+import "./index.scss";
+import { trackTestRuleReportDeleted, trackTestRuleResultClicked } from "../../analytics";
+
+interface TestReportsTableProps {
+  testReports: TestReport[];
+  newReportId: string;
+  isSessionSaving: boolean;
+  refreshTestReports: () => void;
+}
+
+export const TestReportsTable: React.FC<TestReportsTableProps> = ({
+  testReports,
+  newReportId,
+  isSessionSaving,
+  refreshTestReports,
```

Comment:
This components should only receive `testReports`, `deleteReport`. 
Each `testReport` in `testReports` should contain the key for `isSessionSaving` and `isNewReport` on the basis of which this table renders